components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/ConfiguredAuthenticatorsRetrievalClient.java

```diff
@@ -0,0 +1,112 @@
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.mgt.endpoint.util.client;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.owasp.encoder.Encode;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementEndpointUtil;
+import org.wso2.carbon.identity.mgt.endpoint.util.IdentityManagementServiceUtil;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+
+/**
+ * Client which retrieves configure authenticators of an application.
+ */
+public class ConfiguredAuthenticatorsRetrievalClient {
+
+    private static final Log log = LogFactory.getLog(ConfiguredAuthenticatorsRetrievalClient.class);
+    private static final String APPLICATION_API_RELATIVE_PATH = "/api/server/v1/applications";
+    private static final String AUTHENTICATORS = "/authenticators";
+    private static final String SUPER_TENANT = "carbon.super";
+    private static final String CLIENT = "Client ";
+
+    /**
+     * Gets the authenticators configured for an application.
+     *
+     * @param applicationId ID of an application.
+     * @return the list of configured Authenticators
+     * @throws ApplicationDataRetrievalClientException if IO exception occurs when retrieving configured authenticators.
+     */
+    public JSONObject getConfiguredAuthenticators(String applicationId)
+            throws ApplicationDataRetrievalClientException {
+
+        try (CloseableHttpClient httpclient = HttpClientBuilder.create().useSystemProperties().build()) {
+            HttpGet request =
+                    new HttpGet(getApplicationsEndpoint() + "/" + Encode.forUriComponent(applicationId) +
+                            AUTHENTICATORS);
+            setAuthorizationHeader(request);
+
+            try (CloseableHttpResponse response = httpclient.execute(request)) {
```

Comment:
Can't we add this resource into the above try with resource block ?

Reply:
We need to do some more actions in between both blocks